lib/util.js

```diff
@@ -73,6 +73,42 @@ exports.deprecate = function(fn, msg) {
   return deprecated;
 };
 
+// Mark that the synchronous version of a function is deprecated.
+// This is essentially the same as `exports.deprecate()`,
+// except it only warns if the last argument is not a callback function.
+exports.deprecateSync = function(fn, msg) {
```

Comment:
the idea is that if we don't have intention of making function public (and documenting it), it should be marked as private. Since it's used only in crypto I'm not even sure there is a point in defining it in utils.